src/index.ts

```diff
@@ -1,11 +1,12 @@
 import FragmentPlugin from './SwupFragmentPlugin.js';
 export default FragmentPlugin;
+import type { getFragmentVisit } from './inc/functions.js';
 import type { Route, FragmentVisit } from './inc/defs.js';
 export type { Options, Rule, FragmentElement, FragmentVisit } from './inc/defs.js';
 
 declare module 'swup' {
 	export interface Swup {
-		getFragmentVisit?: (route: Route) => FragmentVisit | undefined;
+		getFragmentVisit?: (route: Route) => ReturnType<typeof getFragmentVisit>;
```

Comment:
Could you just do the whole line with `typeof`?

```ts
getFragmentVisit?: typeof getFragmentVisit;
```

Reply:
No, I don't want to expose the logger to users

Reply:
Wait, but maybe I don't need to. Let me look into it.

Reply:
I see, in that case that's the only way I guess?